modules/virt-creating-a-service-from-a-virtual-machine.adoc

```diff
@@ -0,0 +1,196 @@
+the Service  YAML `selector` attribute, which you create later
+in this procedure.
+====
++
+
+[source,yaml]
+----
+aapiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  name: vm-ephemeral
+  namespace: example-namespace
+spec:
+  running: false
+  template:
+    metadata:
+      labels:
+        special: key <1>
+    spec:
+      domain:
+        devices:
+          disks:
+            - name: containerdisk
+              disk:
+                bus: virtio
+            - name: cloudinitdisk
+              disk:
+                bus: virtio
+          interfaces:
+          - masquerade: {}
+            name: default
+        resources:
+          requests:
+            memory: 1024M
+      networks:
+        - name: default
+          pod: {}
+      volumes:
+        - name: containerdisk
+          containerDisk:
+            image: kubevirt/fedora-cloud-container-disk-demo
+        - name: cloudinitdisk
+          cloudInitNoCloud:
+            userData: |
+              #!/bin/bash
+              echo "fedora" | passwd fedora --stdin
+----
+<1> The `special: key` label
++
+
+. Edit the `Service` YAML to create and expose the Service object, connecting by
+specifying the service name. `Service` is scoped by `namespace`.
+.. Specify the `name` for the service you are creating and exposing.
+.. Add a `namespace` key to the `metadata` section of the Service YAML, that corresponds to the
+`namespace` you referenced in the virtual machine YAML.
+.. In the `spec` section of the Service YAML, add `special: key` to the `selector` attribute,
+which corresponds to the `labels` you added in the virtual machine YAML configuration file.
+.. In the `spec` section of the Service YAML, add `type: ClusterIP` for a
+ClusterIP service. Add `targetPort: 22`, exposing the service on SSH port `22`.
++
+To create and expose other types of services externally, outside of the cluster,
+such as `NodePort` and `LoadBalancer`, replace
+`type: ClusterIP` with `type: NodePort` or `type: LoadBalancer`, as appropriate,
+in the Service object definition.
++
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: vmservice <1>
+  namespace: example-namespace <2>
+spec:
+  ports:
+  - port: 27017
+    protocol: TCP
+    targetPort: 22 <3>
+  selector:
+    special: key <4>
+  type: ClusterIP <5>
+----
+<1> Service name
+<2> `namespace` corresponds to the `namespace` you referenced in the virtual machine YAML.
+<3> Exposes SSH `targetPort` 22
+<4> `selector` `special: key` corresponds to the `labels` you added in the virtual machine YAML.
+<5> Defines Service `type: ClusterIP`
+
++
+. Create the `ClusterIP` service using `kubectl`.
++
+
+[source,terminal]
+----
+$ kubectl create -f <service_name>.yaml
+----
+
++
+. Start the virtual machine. If the virtual machine is already running, restart it.
++
+
++
+. Query the Service object to verify it is available and is configured with type `ClusterIP`.
```

Comment:
+1. I was unsure of whether step 7 was verification steps too or not, because it wasn't clear what you were accomplishing by doing them.

So agree to @sjhala-ccs suggestion of either using the `.Verification steps` heading, or if not that, at least make it clear by starting these steps with something like "Verify that <abc> by <xyz>:"

Reply:
This is a task step (Connect to the virtual machine using the service IP and service port.)

There are two expository bullets (and these may be confusing) that we can delete, I think, because the service types are already explained above.

This leaves the two task steps (LIST and CONNECT). List must be done to provide info for the connect command.

Please advise @bergerhoffer, thanks.

Reply:
I think I have resolved this issue. 

The major task step is:
**Connect to the virtual machine using the service IP and service port.**

To accomplish this, there are two sub-steps:
a. List all information.....
b. Connect to the virtual machine.... (Using the SSH command)